frameworks/kotlin/build.gradle

```diff
@@ -0,0 +1,75 @@
+buildscript {
+  ext.kotlin_version = '1.1.3'
+  repositories {
+    jcenter()
+  }
+  dependencies {
+    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+  }
+}
+
+apply plugin: 'kotlin'
+apply plugin: 'application'
+
+mainClassName = System.getenv('KOTLIN_MAIN_CLASS_NAME')
+defaultTasks 'run'
+
+repositories {
+  jcenter()
+}
+
+dependencies {
+  compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+  testCompile 'junit:junit:4.12'
+  testCompile "org.jetbrains.kotlin:kotlin-test-junit:$kotlin_version"
+}
+
+compileKotlin {
+  kotlinOptions {
+  }
+}
+
+test {
+  reports {
+    html.enabled = false
+    junitXml.enabled = false
+  }
+
+  beforeSuite { desc ->
```

Comment:
I decided to split for now because I didn't want to break Java and I also wanted to use these listeners in `build.gradle`. This should allow adding test integrations for other JVM languages easy. I think Groovy and Scala can be supported on top of this by just adding plugins.